src/pointer-analysis/value_set.cpp

```diff
@@ -369,8 +369,9 @@ void value_sett::get_value_set_rec(
 
     const typet &type=ns.follow(expr.op0().type());
 
-    assert(type.id()==ID_array ||
-           type.id()==ID_incomplete_array);
+    INVARIANT(type.id()==ID_array ||
+              type.id()==ID_incomplete_array,
+              "operand 0 of index expression must be an array");
 
```

Comment:
Subjective but I think this might be a DATA_INVARIANT as it is something that should always be true of an index_expr.

Reply:
On reflection, I agree. It's changed in the latest version.